src/ImageHistory.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react';
+import cockpit from 'cockpit';
+import * as utils from './util.js';
+import * as client from './client.js';
+
+import { ListingTable } from "cockpit-components-table.jsx";
+
+const _ = cockpit.gettext;
+
+const IdColumn = Id => {
+    Id = utils.truncate_id(Id);
+    // Not an id but <missing> or something else
+    if (Id.length != 12) {
+        return <div className="pf-u-disabled-color-100">{Id}</div>;
+    }
+    return Id;
+};
+
+const ImageDetails = ({ image }) => {
+    const [history, setHistory] = useState([]);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        client.imageHistory(image.isSystem, image.Id).then(history => setHistory(history))
```

Comment:
you can just pass `setHistory`, no need for the function wrapper.
`client.imageHistory(image.isSystem, image.Id).then(setHistory)`

Reply:
Good point!